src/libs/azure-utils.js

```diff
@@ -1,20 +1,290 @@
+  eastusstg: { flag: '🇺🇸', label: 'East US STG' },
+  northcentralus: { flag: '🇺🇸', label: 'North Central US' },
+  westus: { flag: '🇺🇸', label: 'West US' },
+  jioindiawest: { flag: '', label: 'Jio India West' },
+  centraluseuap: { flag: '🇺🇸', label: 'Central US EUAP' },
+  westcentralus: { flag: '🇺🇸', label: 'West Central US' },
+  southafricawest: { flag: '', label: 'South Africa West' },
+  australiacentral: { flag: '', label: 'Australia Central' },
+  australiacentral2: { flag: '', label: 'Australia Central 2' },
+  australiasoutheast: { flag: '', label: 'Australia Southeast' },
+  japanwest: { flag: '', label: 'Japan West' },
+  jioindiacentral: { flag: '', label: 'Jio India Central' },
+  koreasouth: { flag: '', label: 'Korea South' },
+  southindia: { flag: '', label: 'South India' },
+  westindia: { flag: '', label: 'West India' },
+  canadaeast: { flag: '', label: 'Canada East' },
+  francesouth: { flag: '', label: 'France South' },
+  germanynorth: { flag: '', label: 'Germany North' },
+  norwaywest: { flag: '', label: 'Norway West' },
+  switzerlandwest: { flag: '', label: 'Switzerland West' },
+  ukwest: { flag: '', label: 'UK West' },
+  uaecentral: { flag: '', label: 'UAE Central' },
+  brazilsoutheast: { flag: '', label: 'Brazil Southeast' },
+}
 export const getRegionLabel = key => _.has(key, azureRegions) ? azureRegions[key].label : 'Unknown azure region'
 export const getRegionFlag = key => _.has(key, azureRegions) ? azureRegions[key].flag : '❓'
 
 export const azureMachineTypes = { Standard_DS2_v2: { cpu: 2, ramInGb: 7 }, Standard_DS3_v2: { cpu: 4, ramInGb: 14 }, Standard_DS4_v2: { cpu: 8, ramInGb: 28 }, Standard_DS5_v2: { cpu: 16, ramInGb: 56 } }
 export const getMachineTypeLabel = key => _.has(key, azureMachineTypes) ? `${key}, ${azureMachineTypes[key].cpu} CPU(s), ${azureMachineTypes[key].ramInGb} GBs` : 'Unknown machine type'
 
+// max GB of each azure storage disk type; per https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#standard-ssds
+export const azureDiskTypes = {
+  'E1 LRS': 4,
+  'E2 LRS': 8,
+  'E3 LRS': 16,
+  'E4 LRS': 32,
+  'E6 LRS': 64,
+  'E10 LRS': 128,
+  'E15 LRS': 256,
+  'E20 LRS': 512,
+  'E30 LRS': 1024,
+  'E40 LRS': 2048,
+  'E50 LRS': 4096,
+  'E60 LRS': 8192,
+  'E70 LRS': 16384,
+  'E80 LRS': 32767,
+}
+export const getDiskType = diskSize => _.findKey(maxSize => diskSize <= maxSize, azureDiskTypes)
+
+// Storage prices below are monthly; see https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#standard-ssds
+// Compute prices below are hourly, per https://azure.microsoft.com/en-us/pricing/details/virtual-machines/linux/#pricing.
+// The data below comes from Azure's pricing API: https://prices.azure.com/api/retail/prices
+// There is a script available at https://github.com/broadinstitute/dsp-scripts/blob/master/terra-ui/regionality/parse_pricing_api_azure.py.
+// It parses the API and outputs the response data.
+export const azureRegionToPrices = [
```

Comment:
maybe this was discussed, but why are we hard-coding this rather large block of information here?  Wouldn't it be better to call the source of this information on-demand so that we always get the latest info/rates?

Reply:
This is the pasted-in output from the script introduced in https://github.com/broadinstitute/dsp-scripts/pull/139 - apologies for my oversight in not linking it.

I agree wholeheartedly. Lack of time had me following the existing pattern established for Google pricing: `machines.js` has a big block of code pasted in from `dsp-scripts/terra-ui/regionality/parse_pricing_api.py`'s output, and some hardcoded values for prices and machine configurations.

Another approach is indeed to do this dynamically (i guess from Leo or a connected microservice) based on the user's subscription; the Azure authenticated API does exist for at least some of this. Had I the time to start over before the 25th I'd prefer this approach.

Reply:
Hey @msilva-broad - we are planning on revisiting the cost estimation feature overall. @mlilynolting and I have talked and we agree the current implementation is not the long term approach we want.

Reply:
understood... how about a TODO comment in the code to capture the plan to do things more dynamically.

Reply:
Added! Created https://broadworkbench.atlassian.net/browse/IA-4007 to track.